xml/ntp_banners.xml

```diff
@@ -419,49 +419,247 @@ NTP "banners", taken from a readvar response
-      processor="sun4u", system="SunOS/5.9", leap=00, stratum=2,
+      processor="sun4u", system="SunOS/5.0", leap=00, stratum=2,
     </example>
     <param pos="0" name="os.vendor" value="Sun"/>
     <param pos="0" name="os.family" value="Solaris"/>
     <param pos="0" name="os.product" value="Solaris"/>
     <param pos="0" name="service.family" value="NTP"/>
     <param pos="0" name="service.product" value="NTP"/>
+    <param pos="0" name="os.version" value="2.0"/>
     <param pos="1" name="service.version"/>
     <param pos="2" name="os.arch"/>
-    <param pos="3" name="os.version"/>
   </fingerprint>
-  <fingerprint pattern="^.*processor=&quot;([^ ]+)&quot;,.*system=&quot;SunOS/?([^ ]+)&quot;" flags="REG_DOT_NEWLINE,REG_ICASE">
-    <description>Solaris with no ntp version</description>
+  <fingerprint pattern="^.*version=&quot;ntpd ([^ ]+)[^&quot;]+&quot;,.*processor=&quot;([^ ]+)&quot;,.*system=&quot;SunOS/5.1&quot;" flags="REG_DOT_NEWLINE,REG_ICASE">
+    <description> ntpd running on Solaris 2.1 (SunOS/5.1) </description>
     <example>
-      processor="sun4m", system="SunOS5.6"
+      version="ntpd 4.2.0@1.1161-r Wed Apr 20 11:28:05 EDT 2005 (1)",
+      processor="sun4u", system="SunOS/5.1", leap=00, stratum=2,
     </example>
+    <param pos="0" name="os.vendor" value="Sun"/>
+    <param pos="0" name="os.family" value="Solaris"/>
+    <param pos="0" name="os.product" value="Solaris"/>
+    <param pos="0" name="service.family" value="NTP"/>
+    <param pos="0" name="service.product" value="NTP"/>
+    <param pos="0" name="os.version" value="2.1"/>
+    <param pos="1" name="service.version"/>
+    <param pos="2" name="os.arch"/>
+  </fingerprint>
+  <fingerprint pattern="^.*version=&quot;ntpd ([^ ]+)[^&quot;]+&quot;,.*processor=&quot;([^ ]+)&quot;,.*system=&quot;SunOS/5.2&quot;" flags="REG_DOT_NEWLINE,REG_ICASE">
+    <description> ntpd running on Solaris 2.2 (SunOS/5.2) </description>
     <example>
-      processor="sun4m", system="SunOS5.8"
+      version="ntpd 4.2.0@1.1161-r Wed Apr 20 11:28:05 EDT 2005 (1)",
+      processor="sun4u", system="SunOS/5.2", leap=00, stratum=2,
     </example>
+    <param pos="0" name="os.vendor" value="Sun"/>
+    <param pos="0" name="os.family" value="Solaris"/>
+    <param pos="0" name="os.product" value="Solaris"/>
+    <param pos="0" name="service.family" value="NTP"/>
+    <param pos="0" name="service.product" value="NTP"/>
+    <param pos="0" name="os.version" value="2.2"/>
+    <param pos="1" name="service.version"/>
+    <param pos="2" name="os.arch"/>
+  </fingerprint>
+  <fingerprint pattern="^.*version=&quot;ntpd ([^ ]+)[^&quot;]+&quot;,.*processor=&quot;([^ ]+)&quot;,.*system=&quot;SunOS/5.3&quot;" flags="REG_DOT_NEWLINE,REG_ICASE">
+    <description> ntpd running on Solaris 2.3 (SunOS/5.3) </description>
     <example>
-      processor="sun4u", system="SunOS5.10"
+      version="ntpd 4.2.0@1.1161-r Wed Apr 20 11:28:05 EDT 2005 (1)",
```

Comment:
Thanks Jon, I will incorporate the expected extractions, I have some additional testing to do as well  with these changes before this is ready for merging. Will have a look into your SNMP comment was not aware of those, it is probably worth fixing that while we're here